src/include/containers/List.hpp

```diff
@@ -64,8 +66,88 @@ class List
 		_head = newNode;
 	}
 
+	bool remove(T removeNode)
+	{
+		// base case
+		if (removeNode == _head) {
+			_head = nullptr;
+			return true;
+		}
+
+		for (T node = getHead(); node != nullptr; node = node->getSibling()) {
+			// is sibling the node to remove?
+			if (node->getSibling() == removeNode) {
+				// replace sibling
+				if (node->getSibling() != nullptr) {
+					node->setSibling(node->getSibling()->getSibling());
+
+				} else {
+					node->setSibling(nullptr);
+				}
+
+				return true;
+			}
+		}
+
+		return false;
+	}
+
+	struct Iterator {
+		T node;
+		Iterator(T v) : node(v) {}
+
+		operator T() const { return node; }
+		operator T &() { return node; }
+		T operator* () const { return node; }
+		Iterator &operator++ ()
+		{
+			if (node) {
+				node = node->getSibling();
+			};
+
+			return *this;
+		}
+	};
+
+	Iterator begin() { return Iterator(getHead()); }
+	Iterator end() { return Iterator(nullptr); }
+
 	const T getHead() const { return _head; }
 
+	bool empty() const { return getHead() == nullptr; }
+
+	size_t size() const
+	{
+		size_t sz = 0;
+
+		for (auto node = getHead(); node != nullptr; node = node->getSibling()) {
+			sz++;
+		}
+
+		return sz;
+	}
+
+	void deleteNode(T node)
+	{
+		if (remove(node)) {
+			// only delete if node was successfully removed
+			delete node;
+		}
+	}
+
+	void clear()
+	{
+		auto node = getHead();
+
+		while (node != nullptr) {
+			auto next = node->getSibling();
+			delete node;
+			node = next;
+		}
+
+		_head = nullptr;
+	}
+
```

Comment:
Can we get a unit test for this?

Reply:
Nevermind, found it :smile:

Reply:
Thanks, next up is `Queue` after this merges.